types/block.go

```diff
@@ -572,19 +582,22 @@ func (commit *Commit) ValidateBasic() error {
 		return errors.New("negative Round")
 	}
 
-	if commit.BlockID.IsZero() {
-		return errors.New("commit cannot be for nil block")
-	}
+	// Check that commit.Height is greater than 1
+	// block 1 has commit of block 0 :?? so BlockID & CommitSigs are empty
+	if commit.Height >= 1 {
```

Comment:
This is needed because on `FromProto` we are calling validate basic on the type, so it does not go through the block.validatebasic call.